suite-native/atoms/src/Button/Button.tsx

```diff
@@ -1,18 +1,23 @@
-import { ReactNode, useState } from 'react';
+import { ReactElement, ReactNode, useState } from 'react';
 import { Pressable, PressableProps } from 'react-native';
 import Animated from 'react-native-reanimated';
 
 import { MergeExclusive } from 'type-fest';
+import { G } from '@mobily/ts-belt';
 
 import { Color, TypographyStyle, nativeSpacings } from '@trezor/theme';
 import { NativeStyleObject, prepareNativeStyle, useNativeStyles } from '@trezor/styles';
-import { Icon, IconColor, IconName, IconSize } from '@suite-common/icons';
+import { Icon, IconColor, IconName, IconSize, icons } from '@suite-common/icons';
 
 import { Text } from '../Text';
 import { useButtonPressAnimatedStyle } from './useButtonPressAnimatedStyle';
 import { TestProps } from '../types';
 import { HStack } from '../Stack';
 
+// Using ReactElement instead of ReactNode to exclude string and have type check on IconName
+// and also because string needs to be rendered in the <Text> element anyway
+export type ButtonAccessoryType = IconName | ReactElement | undefined | null;
```

Comment:
Why is `undefined` and `null` needed here?

Reply:
I think that type suffix is redundant here.
```suggestion
export type ButtonAccessory = IconName | ReactElement | undefined | null;
```

Reply:
I put it there to handel undefined | null inside. But i will change it to conditional rendering as you suggested for the readibility

Reply:
0c52c10957

Reply:
the `Type` suffix is still there.

Reply:
good point. updated in 705cf82150